packages/core/aqua-compiler/src/generate/__test__/generate.spec.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2023 Fluence Labs Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { describe, expect, it } from 'vitest';
+import fs from 'fs/promises';
+import { generateTypes, generateSources } from '../index.js';
+import { compileFromPath } from '@fluencelabs/aqua-api';
+import url from 'url';
+import { getPackageJsonContent, PackageJson } from '../../utils.js';
+
+function replacePackageTypes(generated: string, pkg: PackageJson) {
```

Comment:
I got majorly confused here
And now I know why. I think the better name for `generated` - would be `snapshot`

But think a better approach would be to do it differently

- use normal snapshots as you previously did
- before making a snapshot - replace any of the versions generated in the file content to 0.0.0 using e.g. regexp

this way it would be super easy to update snapshots with --updateSnapshots flag
now, with you approach, correct me if I am wrong, you would have to generate snapshots by hand somehow and replace the version with 0.0.0 and put it into snapshots dir yourself

Reply:
Actually this is a great comment - i refactored code a bit. Now i use vitest snapshots instead of making it myself.